src/xenia/app/emulator_window.cc

```diff
@@ -928,6 +937,68 @@ void EmulatorWindow::ToggleFullscreen() {
   SetFullscreen(!window_->IsFullscreen());
 }
 
+void EmulatorWindow::TakeScreenshot() {
+  xe::ui::RawImage image;
+  if (!GetGraphicsSystemPresenter()->CaptureGuestOutput(image) ||
```

Comment:
Note that not all games output the image with the same aspect ratio as the display. Halo 3, for example, presents a 1152x640 image, which has an aspect ratio of 16.2:9 rather than 16:9, so it may appear slightly stretched horizontally. Also, we apply FXAA before writing to the guest output image (for simplicity, as it needs to work at the game's resolution), but CAS or FSR after that — so this will include just the FXAA-processed image with the size provided by the game.